packager/certcache/certcache.go

```diff
@@ -51,6 +51,10 @@ const maxOCSPResponseBytes = 1024 * 1024
 // How often to check if OCSP stapling needs updating.
 const ocspCheckInterval = 1 * time.Hour
 
+// Sentinel value used to communicate that a returned OCSP was fake, and the caller should not attempt to parse it.
+// No, I'm not proud.
+var fakeOCSP = []byte("fake ocsp response")
```

Comment:
It is possible to create a valid OCSP response by using the private key of a self-signed cert such as gateway.go does. But it needs two certs to have the same private key.
https://github.com/ampproject/amppackager/blob/c9993b8ac4d17d1f05d3a1289956dadf3f9c370a/cmd/gateway_server/server.go#L148-L161

Or we can add an additional config to specify the issuer private key file to create a valid OCSP response.

Reply:
Thanks for the pointer; I didn't remember that gateway server did that. However, it seems that the gateway server doesn't fully exercise the certcache code, so it succeeds, whereas my attempt to adopt this fails, with "error validating fake response:  bad OCSP signature: x509: ECDSA verification failure". Can you take a look at https://play.golang.org/p/8owXw00LsRx and see if you can find the error in my code?

Reply:
The error can be resolved with the fix of  https://play.golang.org/p/CvVyUD-_tji . The private key needs to be the issuer key which corresponds to the self-signed cert key in this case.

Reply:
Ah, yes, that makes perfect sense. Thanks for the bug fix; I've updated this to generate a GOOD OCSP response as suggested.